src/api/dsl/Expressions.h

```diff
@@ -476,9 +478,11 @@ class ScriptExpression : public Expression {
 // So here - we will implement this first. When necessary, we may want to introduce different expresssion type.
 // TODO(cao) - need rework this since we need to come up with a framework
 // to allow customized UDAFs to be plugged in
+struct void_type { using type = uint64_t; };
```

Comment:
hacky change, basically I want to get the inputtype of args, any suggestions?

Reply:
Add a concrete UDF Expression class `HistExpression` is probably a good idea like others, similar like InExpression.

In the future, if we want to use UDFExpression to support serialization of any varargs, we should consider introducing a Param object to wrap each arg with both Type and Value for serde, this is definitely a complex approach, so may not a good option for now.

Reply:
thanks! created a class by using UDFExpression as base class